Flickr/Sources/FlickrAppVCs/FlickrApiCoordinator.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  URLCoordinator.swift
+//  Flickr
+//
+//  Created by Ananth Desai on 17/03/22.
+//
+
+import Foundation
+
+class ApiCoordinator {
+    struct Photos: Codable {
+        var photos: PhotoArray
+    }
+
+    struct PhotoArray: Codable {
+        var photo: [SinglePhoto]
+    }
+
+    struct SinglePhoto: Codable {
+        var id: String
+        var owner: String
+        var secret: String
+        var server: String
+        var title: String
+    }
+
+    private func returnSearchUrl(searchString: String) -> URL? {
+        URL(string: "\(baseSearchUrl)/?method=\(method)&api_key=\(apiKey)&format=\(format)&nojsoncallback=\(noJsonCallback)&text=\(searchString)")
+    }
+
+    private func returnImageURl(image: SinglePhoto) -> URL? {
+        URL(string: "\(imageSearchUrl)/\(image.server)/\(image.id)_\(image.secret).jpg")
+    }
+
+    func fetchPhotos(searchString: String) -> Bool {
+        guard let url = returnSearchUrl(searchString: searchString) else {
+            return false
+        }
+        let task = URLSession.shared.dataTask(with: url, completionHandler: { data, _, error in
+            guard let data = data, error == nil else {
+                return
+            }
+            var result: Photos?
+            do {
+                result = try JSONDecoder().decode(Photos.self, from: data)
+            } catch {
+                return
+            }
+            guard let result = result else {
+                return
+            }
+            guard self.constructIndividualUrls(result) == true else {
+                return
+            }
+        })
+        task.resume()
+        return true
+    }
```

Comment:
Why are you returning a `bool` here ?

Reply:
This file has been removed. The updated code is in the branch "ananth/task4"

Reply:
@Ananth-Desai  But I can't merge this PR by taking a look at some other PR. Please update this PR first.

Reply:
I have made the changes to this file in this branch